components/Header.tsx

```diff
@@ -1,11 +1,105 @@
 import Link from "next/link";
+import Image from "next/image";
+import { useRouter } from 'next/router';
+import { useEffect, useState } from "react";
 
 const Header = () => {
+    const router = useRouter();
+    const currentPath = router.pathname;
+
+    const isActive = (path: string): string => {
+        return currentPath === path ? 'border-project-green bg-project-green font-bold' : 'group';
+    };
+
+    const [toggleMenu, setToggleMenu] = useState(false);
+
+    useEffect(() => {
+        const handleResize = () => {
+        //   setWindowWidth(window.innerWidth);
+          if (window.innerWidth > 1023) {
+            setToggleMenu(false)
+          }
+        };
+    
+        // Set the initial window width
+        // setWindowWidth(window.innerWidth);
+    
+        // Attach event listener for window resize
+        window.addEventListener('resize', handleResize);
+    
+        // Clean up the event listener on component unmount
+        return () => {
+          window.removeEventListener('resize', handleResize);
+        };
+      }, []);
+
+    function setMenuState ():any {
```

Comment:
We don't need the `any` type here. This function doesn't return anything, hence this is a `void` type. Same as before, hover `setMenuState` after removing `any` to see TypeScript already knows that is a `void` type